components/brave_ads/core/internal/user/user_interaction/ad_events/ad_event_handler_util.h

```diff
@@ -17,6 +19,49 @@ bool HasFiredAdEvent(const AdInfo& ad,
                      const AdEventList& ad_events,
                      const ConfirmationType& confirmation_type);
 
+template <typename T>
+bool WasAdServed(const AdInfo& ad,
+                 const AdEventList& ad_events,
+                 const T& event_type) {
+  CHECK(mojom::IsKnownEnumValue(event_type));
```

Comment:
You most probably don't need these `CHECK`s on `mojom::IsKnownEnumValue`, unless you explicitly turn off Mojom validation for these enums somehow (https://chromium.googlesource.com/chromium/src/+/HEAD/mojo/public/tools/bindings/README.md#enums).

Reply:
Thanks, will follow up in a separate issue/PR.